test/uk/gov/hmrc/iossregistration/services/SaveForLaterServiceSpec.scala

```diff
@@ -0,0 +1,126 @@
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.iossregistration.connectors.GetVatInfoConnector
+import uk.gov.hmrc.iossregistration.generators.Generators
+import uk.gov.hmrc.iossregistration.models.{NotFound, SavedUserAnswers}
+import uk.gov.hmrc.iossregistration.models.des.VatCustomerInfo
+import uk.gov.hmrc.iossregistration.models.requests.{SaveForLaterRequest, SaveForLaterResponse}
+import uk.gov.hmrc.iossregistration.repositories.SaveForLaterRepository
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import java.time.{Clock, Instant, ZoneId}
+import scala.concurrent.Future
+
+
+class SaveForLaterServiceSpec
+  extends AnyFreeSpec
+    with Matchers
+    with MockitoSugar
+    with ScalaCheckPropertyChecks
+    with Generators
+    with OptionValues
+    with ScalaFutures {
+
+  implicit private lazy val hc: HeaderCarrier = HeaderCarrier()
+
+  ".saveAnswers" - {
+
+    "must create a SavedUserAnswers, attempt to save it to the repository, and respond with the result of saving" in {
+
+      val now = Instant.now
+      val stubClock = Clock.fixed(now, ZoneId.systemDefault())
+      val answers = arbitrary[SavedUserAnswers].sample.value
+      val insertResult = answers
+      val mockSaveForLaterRepository = mock[SaveForLaterRepository]
+      val mockGetVatInfoConnector = mock[GetVatInfoConnector]
+
+      when(mockSaveForLaterRepository.set(any())) thenReturn Future.successful(insertResult)
+
+      val request = arbitrary[SaveForLaterRequest].sample.value
+      val service = new SaveForLaterService(mockSaveForLaterRepository, mockGetVatInfoConnector, stubClock)
+
+      val result = service.saveAnswers(request).futureValue
+
+      result mustEqual insertResult
+      verify(mockSaveForLaterRepository, times(1)).set(any())
+      verifyNoInteractions(mockGetVatInfoConnector)
+    }
+  }
+
+  ".get" - {
+
+    "must retrieve a Saved User Answers record when it is found with a VatCustomerInfo" in {
+      val now = Instant.now
+      val stubClock = Clock.fixed(now, ZoneId.systemDefault())
+      val answers = arbitrary[SavedUserAnswers].sample.value
+      val vatInfo = arbitrary[VatCustomerInfo].sample.value
+      val mockRepository = mock[SaveForLaterRepository]
+      val mockGetVatInfoConnector = mock[GetVatInfoConnector]
+
+      val vrn = arbitrary[Vrn].sample.value
+
+      when(mockRepository.get(any())) thenReturn Future.successful(Some(answers))
+      when(mockGetVatInfoConnector.getVatCustomerDetails(any())(any())) thenReturn Future.successful(Right(vatInfo))
+
+      val service = new SaveForLaterService(mockRepository, mockGetVatInfoConnector, stubClock)
+
+      val result = service.get(vrn).futureValue
+      result mustBe Some(SaveForLaterResponse(answers, vatInfo))
+      verify(mockRepository, times(1)).get(vrn)
+      verify(mockGetVatInfoConnector, times(1)).getVatCustomerDetails(answers.vrn)
+
+    }
+
+    "must fail if a VatCustomerInfo cannot be retrieved but a saved answers can" in {
```

Comment:
VatCustomerInfo should always be available if we have a saved answer, returning none would hide problems